openedx/openedx-platform
# Use appropriate URL scheme for bulk emails

Uses http or https appropriately in bulk email urls, and adds unit tests for affected methods.

**Sandbox URL**:

This sandbox uses `LMS_ROOT_URL: http://pr12934.sandbox.opencraft.hosting/`:
- LMS: http://pr12934.sandbox.opencraft.hosting/
- Studio: http://studio-pr12934.sandbox.opencraft.hosting/

To test this feature with a sandbox configured with https root url (but still not using SSL), add this to your `/etc/hosts`, and use the above http URL.

```
149.202.181.19  pr12934.sandbox.opencraft.hosting
```

**Testing instructions**:

Manual testing:
1. Setup edxapp service.  Note that in the devstack, [`LMS_ROOT_URL` uses "http"` by default](https://github.com/edx/edx-platform/blob/550f8520b4be6990db5c73919983e18bed2fbad0/cms/envs/dev.py#L63), so to use https, set: 
   
   ```
   EDXAPP_ENV_EXTRA:
        LMS_ROOT_URL: "https://localhost:8000"
   ```
2. Enable Bulk Emails via /admin/bulk_email/bulkemailflag/add : `Enabled = Yes`, `Require course email auth = No`
3. Register and enrol as student with your own email address.
4. As instructor, send a bulk email to all enrolled students via the Instructor Dashboard.
5. Check your email (it might take a few minutes, and might be in your spam folder).  Verify that the links in the footer use http or https as appropriate for your edxapp service.

Run automated tests:
1. Set up devstack
2. Run `paver test_system -s lms --fasttest --disable-migrations -t lms/djangoapps/bulk_email/tests/test_email.py`
3. <strike>Run `paver test_system -s lms --fasttest --disable-migrations -t lms/djangoapps/instructor_task/tests/test_api.py`</strike> (test_api.py is now unchanged)

**Reviewers**
- [x] edX reviewer @jcdyer

**Settings**

``` yaml
EDXAPP_ENV_EXTRA:
  LMS_ROOT_URL: 'https://pr12934.sandbox.opencraft.hosting/'
```
